pootle/apps/pootle_store/unit/timeline.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from hashlib import md5
+from itertools import groupby
+
+from django.contrib.auth import get_user_model
+from django.core.urlresolvers import reverse
+from django.db.models import Q
+from django.utils.html import format_html
+from django.utils.safestring import mark_safe
+from django.utils.translation import ugettext as _
+
+from pootle_misc.checks import check_names
+from pootle_statistics.models import (
+    Submission, SubmissionFields, SubmissionTypes)
+
+from pootle_store.fields import to_python
+from pootle_store.util import STATES_MAP
+
+
+class ProxyTimelineUser(object):
+
+    def __init__(self, submission):
+        self.submission = submission
+
+    @property
+    def display_name(self):
+        return (
+            self.submission["submitter__full_name"].strip()
+            if self.submission["submitter__full_name"].strip()
+            else self.submission["submitter__username"])
+
+    def get_absolute_url(self):
+        return reverse(
+            'pootle-user-profile',
+            args=[self.submission["submitter__username"]])
+
+    def gravatar_url(self, size=80):
+        email_hash = md5(self.submission['submitter__email']).hexdigest()
```

Comment:
I understand these proxy classes are for performance reasons (and to avoid filling up the memory with heavy `User` models), although it'd be cool to avoid duplicated logic with regards to the `User` model by externalizing the logic as functions and having both the proxy classes and models use these. Thoughts?